### ISSUE TITLE
Error with hparams.proximity_bias=True

### ISSUE THREAD

**Issue by Leechung** [tensorflow/tensor2tensor#1262]
### Description
Error in common_attention.attention_bias_to_padding with setting `hparams.proximity_bias=True`
In the transformer_layers.transformer_encoder.transformer_encoder line 152,  the `operation padding = common_attention.attention_bias_to_padding`  performs `tf.squeeze(x, [1,2])`, but attention_bias tensor returned with proximity_bias=True is [batch_size, 1, seq_len, seq_len] and not [batch_size, 1, 1, seq_len] when proximity_bias=False. 

Not sure how effective proximity_bias is, but if it is there probably it has to be working. 

https://github.com/tensorflow/tensor2tensor/blob/d2b6b3a0885dcba995d74fe97f33c2e4b5ce2cf8/tensor2tensor/layers/common_attention.py#L919 

I am fixing it by changing the above line to this

`return tf.squeeze(tf.to_float(tf.less(tf.reduce_sum(attention_bias, 2,keepdims=True), -1)), axis=[1, 2])`

### For bugs: reproduction and error logs

```
#1 : set hparams.proximity_bias=True
...
```

```
# Error logs:
InvalidArgumentError (see above for traceback): Can not squeeze dim[2], expected a dimension of 1, got 832
...
```
